Mathlib/NumberTheory/FactorisationProperties.lean

```diff
@@ -0,0 +1,208 @@
+
+theorem prime_not_abundant (h : Prime n) : ¬ Abundant n :=
+  fun h1 ↦ (h.one_lt.trans h1).ne' (sum_properDivisors_eq_one_iff_prime.mpr h)
+
+theorem prime_not_weird (h : Prime n) : ¬ WeirdNumber n := by
+  simp only [Nat.WeirdNumber, not_and_or]
+  left
+  exact prime_not_abundant h
+
+theorem prime_not_pseudoperfect (h : Nat.Prime n) : ¬ Nat.Pseudoperfect n := by
+  rw [not_pseudoperfect_iff_forall,
+    show powerset (properDivisors n) = {∅, {1}} by rw [Prime.properDivisors h]; rfl]
+  refine Or.inr (fun s hs ↦ ?_)
+  fin_cases hs <;>
+  simp only [sum_empty, sum_singleton] <;>
+  linarith [Prime.one_lt h]
+
+theorem prime_not_perfect (h : Prime n) : ¬ Perfect n := by
+  have h1 := prime_not_pseudoperfect h
+  revert h1
+  exact not_imp_not.mpr (perfect_pseudoperfect)
+
+/-- Any natural number power of a prime is deficient -/
+theorem prime_pow_deficient (h : Prime n) : Deficient (n ^ m) := by
+  rcases Nat.eq_zero_or_pos m with (hL | _)
+  · rw [hL, Nat.pow_zero]
+    exact one_deficient
+  · have h1 : properDivisors (n ^ m) = image (n ^ ·) (range m) := by
+      apply subset_antisymm <;> intro a
+      · simp only [mem_properDivisors, mem_image, mem_range, dvd_prime_pow h]
+        rintro ⟨⟨t, ht, rfl⟩, ha'⟩
+        exact ⟨t, lt_of_le_of_ne ht (fun ht' ↦ lt_irrefl _ (ht' ▸ ha')), rfl⟩
+      · simp only [mem_image, mem_range, mem_properDivisors, forall_exists_index, and_imp]
+        intro x hx hy
+        constructor
+        · rw [← hy, dvd_prime_pow h]
+          exact ⟨x, Nat.le_of_succ_le hx, rfl⟩
+        · rw [← hy]
+          exact (Nat.pow_lt_pow_iff_right (Prime.two_le h)).mpr hx
+    have h2 : ∑ i in image (fun x => n ^ x) (range m), i = ∑ i in range m, n^i := by
+      rw [Finset.sum_image]
+      rintro x _ y _ hnxy
+      by_contra hc
+      rcases (Ne.lt_or_lt hc) with hx | hx <;>
+      linarith [(pow_lt_pow_iff_right (Prime.one_lt h)).mpr hx]
+    rw [Deficient, h1, h2]
+    calc
+      ∑ i ∈ range m, n ^ i
+        = (n ^ m - 1) / (n - 1) := (Nat.geomSum_eq (Prime.two_le h) _)
+      _ ≤ (n ^ m - 1) := Nat.div_le_self (n ^ m - 1) (n - 1)
+      _ < n ^ m := sub_lt (pow_pos (Prime.pos h) m) (Nat.one_pos)
+
+theorem prime_deficient (h : Prime n) : Deficient n := by
+  rw [← pow_one n]
+  exact prime_pow_deficient h
+
+/-- There exists infinitely many deficient numbers -/
+theorem exists_infinite_deficients : ∃ d, n ≤ d ∧ Deficient d := by
+  obtain ⟨p, ⟨h1, h2⟩⟩ := exists_infinite_primes n
+  exact ⟨p, h1, prime_deficient h2⟩
+
+theorem exists_infinite_even_deficients : ∃ d, n ≤ d ∧ Deficient d ∧ Even d := by
+  use 2 ^ (n + 1)
+  constructor
+  · calc
+      n ≤ 2 ^ n := Nat.le_of_lt (lt_two_pow n)
+      _ ≤ 2 ^ (n + 1) := le_of_succ_le ((Nat.pow_lt_pow_iff_right (Nat.one_lt_two)).mpr
+        (lt_add_one n))
+  · exact ⟨prime_pow_deficient prime_two, even_pow.mpr ⟨even_two, Ne.symm
+      (NeZero.ne' (n + 1))⟩⟩
+
+theorem exists_infinite_odd_deficients : ∃ d, n ≤ d ∧ Nat.Deficient d ∧ Odd d := by
```

Comment:
Probably it's less painful here to just use the fact that primes > 2 are odd.

Reply:
Good point. I'll implement that.

Reply:
How do I let lean know that I'm going to use a Prime number greater than two?